main.go

```diff
@@ -356,24 +355,69 @@ func main() {
 				Certificates:             []tls.Certificate{cert},
 			}
 
-			log.WithField("address", listener).
+			log.WithField("address", address).
 				Info("listening on TLS address")
 
-			lsnr, err := tls.Listen("tcp", listener, server.TLSConfig)
-			if err != nil {
-				log.Fatal(err)
-			}
-			defer lsnr.Close()
+			listener := listenOnTLSAddr(server, address, server.TLSConfig)
+			listeners = append(listeners, listener)
 
-			go server.Serve(lsnr)
 		} else {
-			log.WithField("address", listener).
+			log.WithField("address", address).
 				Fatal("unknown protocol in address")
 		}
 	}
 
-	// TODO: handle SIGTERM and gracefully shutdown
-	for true {
-		time.Sleep(time.Minute)
+	handleSignals(listeners)
+}
+
+func listenOnAddr(server *smtpd.Server, addr string) net.Listener {
+	listener, err := net.Listen("tcp", addr)
+	if err != nil {
+		log.WithField("err", err).
+			WithField("address", addr).
+			Fatal("could not listen on address")
+	}
+
+	go server.Serve(listener)
+	return listener
+}
+
+func listenOnTLSAddr(server *smtpd.Server, addr string, config *tls.Config) net.Listener {
+	listener, err := tls.Listen("tcp", addr, config)
+	if err != nil {
+		log.WithField("err", err).
+			WithField("address", addr).
+			Fatal("could not listen on address")
 	}
+
+	go server.Serve(listener)
+	return listener
+}
+
+func handleSignals(servers []net.Listener) {
```

Comment:
nit: let's add godoc on the new functions that we added


note: comments that start with `nit:` are nitpicks, and can be ignored :)

Reply:
One usually only adds godocs for exported functions, plus none of the other functions in this file are commented; I'm hoping that the intent is expressed clearly by the naming so that we don't need comments.

What do you think?

Reply:
makes sense 👍🏼